core/src/main/kotlin/api/OrganizationsRoute.kt

```diff
@@ -78,15 +80,15 @@ fun Route.organizations() = route("organizations") {
     val infrastructureServiceService by inject<InfrastructureServiceService>()
 
     get(getOrganizations) { _ ->
-        requireSuperuser()
-
         val pagingOptions = call.pagingOptions(SortProperty("name", SortDirection.ASCENDING))
 
-        val organizations = organizationService.listOrganizations(pagingOptions.mapToModel())
-        val pagedResponse = PagedResponse(
-            organizations.map { it.mapToApi() },
-            pagingOptions
-        )
+        val organizations = organizationService
+            .listOrganizations(pagingOptions.copy(limit = null, offset = null).mapToModel())
+            .filter { hasPermission(it.id, OrganizationPermission.READ) }
+            .paginate(pagingOptions)
```

Comment:
Will this pagination in memory work in all cases? I am not sure how the database behaves if no sort order is specified; but there is probably no guarantee that the order of results is always the same? Maybe, to be safe, always a sort order needs to be specified?